modules/library/main/src/main/java/org/geotools/geometry/jts/GeometryClipper.java

```diff
@@ -76,6 +79,49 @@ public GeometryClipper(Envelope bounds) {
     }
     
     /**
+     * This will try to handle failures when clipping - i.e. because of invalid input geometries (often 
+     * caused by simplification).
+     * 
+     * This attempts to do a normal clip().  If it fails, it will try to do more to ensure the clip 
+     * works properly.  
+     * 
+     * The first attempt to correct the geometry is to put the points on a precision grid and redo
+     * the clip. 
+     * If this fails, the original geometry is returned.
+     * 
+     * see {@link #clip(Geometry, boolean) clip}
+     * 
+     * @param g
+     * @param ensureValid
+     * @param scale 0=double precision Precision Model
+     * @return
+     */
+    public Geometry clipFailResistant(Geometry g, boolean ensureValid, double scale) {
```

Comment:
I did a quick test.  For my polygon, doing a buffer(0) makes it valid.  I'll add this in.